### ISSUE TITLE
Investigate latest failures (NumPy 1.10?)

### ISSUE THREAD

**Issue by Juanlu001** [poliastro/poliastro#73]

**Comment by newlawrence** [comment]
This is what I found out so far.

I've modified the scripts to build **Poliastro** using specific versions of **Numpy** and yes, it seems that there's some kind of issue arising with version 10 (here are the **OS X** logs from [Travis](https://travis-ci.org/newlawrence/poliastro/builds/94829872) & **Windows logs** from [Appveyor](https://ci.appveyor.com/project/newlawrence/poliastro/build/0.4.0.dev0-6)).

But it seemed that everything was OK for **Anaconda Cloud**, doesn't it?

Well, the problem is that **Anaconda Cloud** uses the **conda's build** system to build and test the packages and there're big troubles to tell it to build against specific **Numpy** versions.
If you read the logs, a package ***poliastro-0.4.0.dev0-py27_1*** is generated, while if it also was dependent on **Numpy**'s version it should be ***poliastro-0.4.0.dev0-np19py27_1***.

One of the dependencies of **Poliastro** must be incompatible with **Numpy 1.10** as **conda build** always chooses **Numpy 1.9**, so if we apply the following [patch](https://github.com/newlawrence/poliastro/commit/a8a2150fd873b7b34b51d03c0fe57a466b7d066f) to force the specific **Numpy** version, we obtain the same results for **Linux** from [Anaconda Cloud](https://anaconda.org/newlawrence/poliastro/builds/13) that we got from **Travis** and **Appveyor**.

**How the path works:** **conda build** relies on the **CONDA_PY** and **CONDA_NPY** environment variables to build the packages using specific versions of **Python** and **Numpy**. When both of them are set up, the resulting package will be named ***package-version-npYYpyXX_bnum***. If they're not set (neither **--python** nor **--numpy** flags are provided), **conda build** will build the package using the current **Python** version installed and the most recent version of **Numpy** allowed by the dependancies. As **Anaconda Cloud** doesn't set up the **CONDA_NPY** variable, we do it manually. The changes made to the **meta.yaml** ensure the package is built using the specified **Numpy** version.

In summary, **Poliastro** seems that must be built against a specific **Numpy** version due to its dependencies, and currently there's some kind of incompatibility with **Numpy 1.10** as we can see in [1](https://anaconda.org/newlawrence/poliastro/builds/13), [2](https://travis-ci.org/newlawrence/poliastro/builds/94829872) and [3](https://ci.appveyor.com/project/newlawrence/poliastro/build/0.4.0.dev0-6).

**PS**: you must provide a **conda jplephem** package for **Python 3.5**. If you want, I can submit a pull request only for **Numpy 1.9** while the problem with the version 10 is solved, this will allow the people to safely use the package with the old version in the mean time.

**Comment by Juanlu001** [comment]
Thanks a lot for the legwork @newlawrence! You are right: the dependency on NumPy is quite strong and requires building packages for each version independently. However, the use of the `CONDA_NPY` variable has its quircks, see this issue that I reported a few weeks ago:

https://github.com/conda/conda-build/issues/648

The point is that this process is still subject to change in conda-build, see these two issues:

https://github.com/conda/conda-build/pull/573
https://github.com/conda/conda-build/pull/650

Right now I would prefer a simple or even manual process to build the packages, because there's no point in automating something that will change within a month. I don't mind manually building two packages: one against NumPy 1.9 and another one against NumPy 1.10.

**Comment by Juanlu001** [comment]
On the other hand: half of your builds are failing because `'poliastro' has unsatisfiable dependencies`. I would bet the other half fail because of astropy, but I didn't find the specific bug yet - I'm sure they already know.

**Comment by newlawrence** [comment]
Yeah I'm aware of that, it's a big mess, they're having so much trouble with **anaconda-client** and **conda-build**.

The pull request I suggest will ensure you can make the tests specifying the **Numpy** version, soy you can safely have your **master** branch on green using 1.9 while the problem with 1.10 is solved. I think we can do the tests in **Anaconda Cloud** without building the package so we can forget about **CONDA_NPY**, I'll give it a try.

**Comment by newlawrence** [comment]
I forgot, I was aware of the problem with **CONDA_NPY** and as I told above, I made this work around to the **meta.yaml** file:
```
    - numpy 1.9*  # [np == 19]
    - numpy 1.10* # [np == 110]
```
I know is monkey patching, but I've been using it and it works. It's a solution while all of that mess with **--numpy** flag is solved.

**Comment by Juanlu001** [comment]
Funny thing: did you notice the error message? `module compiled against API version a but this version of numpy is 9`. It's using hex to print NumPy versions, see https://github.com/numpy/numpy/pull/6608.

**Comment by Juanlu001** [comment]
Per @newlawrence I relaunched a failing job in Travis CI after merging #76 and now it's clear what is happening:

https://travis-ci.org/poliastro/poliastro/jobs/93108122

It's safe to say that I can fix these errors this weekend.

**Comment by newlawrence** [comment]
Happy to read it! Remember to [fix](https://github.com/newlawrence/poliastro/commit/8654dbc26c27858823da008486956384c51c939f) the **Travis** and **Appveyor** scripts.

**Comment by Juanlu001** [comment]
Well, it seems matplotlib is broken on Anaconda. https://github.com/ContinuumIO/anaconda-issues/issues/573
